control-plane/connect-inject/webhook/mesh_webhook_test.go

```diff
@@ -12,6 +12,7 @@ import (
 	mapset "github.com/deckarep/golang-set"
 	logrtest "github.com/go-logr/logr/testr"
 	"github.com/hashicorp/consul-k8s/control-plane/connect-inject/constants"
+	"github.com/hashicorp/consul-k8s/control-plane/connect-inject/lifecycle"
```

Comment:
**thought:** Maybe consult with Thomas on this one. I am surprised that we do not have a webhook test that looks at the actual mutated pod itself?

I was thinking that a test could be added that is very similar to `TestHandlerHandle` with your initial pod and then instead of `Patches` you compare against a pod spec.

I won't block on this idea though..

Reply:
That is a super fair point. I'll make a note of that to come back and write that unit test in the future.